Neos.ContentRepositoryRegistry/Classes/Configuration/NodeTypeEnrichmentService.php

```diff
@@ -358,9 +368,9 @@ protected function getPropertyLabelTranslationId($nodeTypeSpecificPrefix, $prope
      * @param string $labelPath
      * @return string
      */
-    protected function getPropertyConfigurationTranslationId($nodeTypeSpecificPrefix, $propertyName, $labelPath)
+    protected function getConfigurationTranslationId(string $nodeTypeSpecificPrefix, string $propertyName, string $labelPath, string $configurationType): string
     {
-        return $nodeTypeSpecificPrefix . 'properties.' . $propertyName . '.' . $labelPath;
+        return $nodeTypeSpecificPrefix . $configurationType . '.' . $propertyName . '.' . $labelPath;
```

Comment:
sure that was the next logical step but its absolute maniac to have to read this one 😅 ... but i dont know if the alternative would be better.
I guess the real first step to improve the code here would be to look away and write tests for it because as far as i know there are 0 :D

Reply:
before there are way too many strings id like to propose to create an enum for `string $configurationType` but only if we dont find anything better.
Otherwise the code is impossible to static analyse and also not easy to read / trust.

Reply:
Using enum would be nicer for properties/references 👍 

Only problem there would be how to handle reference edge properties?
My proposed suggestion was to add the "parent" property name there: 

```  
references:
    metaNavigationItems:
      type: references

      properties:
        text:
          type: string
          ui:
            label: i18n
```
in the new code uses the label 
`Neos.Demo:NodeTypes.Document.Page:metaNavigationItems.properties.text`

Then we would have to add another optional $parentPropertyName to the function, which does not really make it easier to understand - even with enum. 

We could also use 
`Neos.Demo:NodeTypes.Document.Page:references.metaNavigationItems.properties.text` 
then enum would work again but we would still have to pass $parentPropertyName to the function

Reply:
Ah i see now i understand what the code does thanks ;) I didnt think of reference properties till now ^^
... and then we have to ask ourselves if we really want to keep that `i18n` magic forever.
As said in https://github.com/neos/neos-development-collection/issues/2867#issuecomment-575714747 its currently not easy for us to implement (as we cannot use a post processor for it) and the question is if we like to keep this feature for the new way of declaring references or can think of something better and more transparent?

Reply:
We could also leave out the reference properties for now, use enum for the properties/references and then make that a future-us problem. But it would be a soonish-problem since we do have to translate those at some point

Reply:
@mhsdesign Tests are added 👍